leetcode/numbers_2.rs

```diff
@@ -0,0 +1,21 @@
+//Subtract Sum from Product
+pub fn subtract_product_and_sum(n: i32) -> i32 {
+    let mut stringn = n.to_string();
+    let mut charn = stringn.chars();
+
+    let mut vecn = charn.map(|x| x.to_digit(10).unwrap() as i32);
+    let mut product: i32 = vecn.clone().product();
```

Comment:
Can you explain why `clone` is needed here?

Reply:
Because vecn was moved as it was called to line 7, and in the next line it was used again after it was used. So I had to clone it because of ownership issues.